what4/test/Bool.hs

```diff
@@ -0,0 +1,280 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE ViewPatterns #-}
+
+module Bool where
+
+import Control.Monad (unless, when)
+import Control.Monad.IO.Class (MonadIO, liftIO)
+import Control.Monad.State.Strict qualified as State
+import Control.Monad.Trans (lift)
+import Data.Coerce (coerce)
+import Data.Either (isRight)
+import Data.Foldable (traverse_)
+import Data.Map qualified as Map
+import Data.Parameterized.Map qualified as MapF
+import Data.Parameterized.Nonce (newIONonceGenerator)
+import Data.Parameterized.Some (Some(Some))
+import Hedgehog (GenT)
+import Hedgehog.Gen qualified as Gen
+import Hedgehog.Internal.Gen qualified as HG
+import Hedgehog.Internal.Property qualified as HG
+import Test.Tasty.Hedgehog qualified as THG
+import Test.Tasty qualified as T
+import What4.Expr.BoolMap qualified as BM
+import What4.Expr.Builder
+import What4.Expr (EmptyExprBuilderState(EmptyExprBuilderState))
+import What4.Interface
+
+data BExpr a
```

Comment:
Worth a comment here saying what the purpose of this `BExpr` data type is. My understanding is that it is meant to provide a simple reference implementation of how boolean evaluation should work that we can compare against the more complication evaluation scheme in `what4` itself. Is this correct?

Reply:
Good call, done!